moveit_ros/perception/handeye_calibration_target/include/moveit/handeye_calibration_target/handeye_target_base.h

```diff
@@ -0,0 +1,190 @@
+   * @param dictionary_id Marker dictionary id.
+   * @return True if parameters are correctly set, false otherwise.
+   */
+  virtual bool setTargetIntrinsicParams(const int& markers_x, const int& markers_y, const int& marker_size_,
+                                        const int& separation, const int& border_bits,
+                                        const std::string& dictionary_id) = 0;
+
+  /**
+   * @brief Set the parameters of printed target, for updating the real target parameters.
+   * @param marker_measured_size Printed marker size.
+   * @param marker_measured_separation Printed marker separation distance.
+   * @return True if parameters are correctly set, false otherwise.
+   */
+  virtual bool setTargetDimension(const double& marker_measured_size, const double& marker_measured_separation) = 0;
+
+  /**
+   * @brief Create an target image, so that the target can be viewed and printed.
+   * @param image Use for storing the created image.
+   * @return True if no errors happen, false otherwise.
+   */
+  virtual bool createTargetImage(cv::Mat& image) = 0;
+
+  /**
+   * @brief Given an image containing a target captured from a camera view point, get the target pose with respect to
+   * the camera optical frame. Target parameters and camera intrinsic parameters should be correctly set
+   * before calling this function.
+   * @param image Input image, assume a grayscale image.
+   * @return True if no errors happen, false otherwise.
+   */
+  virtual bool detectTargetPose(cv::Mat& image) = 0;
+
+  /**
+   * @brief Get available target dictionary ids that can be used to fill the dropdown list of the handeye calibration
+   * GUI component.
+   * @return A vector of dictinary names.
+   */
+  virtual std::vector<std::string> getDictionaryIds() = 0;
+
+  /**
+   * @brief Get `TransformStamped` message from the target detection result, use for TF publish.
+   * @param frame_id The name of the frame this transform is with respect to.
+   * @return A `TransformStamped` message.
+   */
+  virtual geometry_msgs::TransformStamped getTransformStamped(std::string& frame_id) = 0;
+
+  /**
+   * @brief Set camera intrinsic parameters, e.g. camera intrinsic matrix and distortion coefficients.
+   * @param msg Input camera info message.
+   * @return Ture if the input camera info format is correct, false otherwise.
+   */
+  virtual bool setCameraIntrinsicParams(const sensor_msgs::CameraInfoPtr& msg)
+  {
+    if (msg)
+    {
+      ROS_DEBUG_STREAM_NAMED("handeye_target_base", "CameraInfo set: " << *msg);
+
+      if (msg->K.size() == 9 && msg->D.size() == 5)
+      {
+        std::lock_guard<std::mutex> lck(base_mtx_);
+        // Store camera matrix info
+        for (size_t i = 0; i < 3; i++)
+        {
+          for (size_t j = 0; j < 3; j++)
+          {
+            camera_matrix_.at<double>(i, j) = msg->K[i * 3 + j];
+          }
+        }
+
+        // Store camera distortion info
+        for (size_t i = 0; i < 5; i++)
```

Comment:
create const for 5... what is that?

Reply:
Defined [this](https://github.com/ros-planning/moveit/pull/1558/files#diff-02e9ff8e0911fa44acaf644c832988f7R46) to const as well.